### PR TITLE
Protips are remembered in json now instead of your OS's registry

### PR DESCRIPTION
**Pull request by MaxIsJoe**
CL: [New] - You can forget all protips to make them appear again from the help menu -> settings now.
CL: [Fix] - Fixed an issue where the "Remember" button would only appear on widescreen resolutions.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Learning/ProtipManager.cs [hunk 1]
```diff
@@ -39,13 +45,53 @@ public override void Awake()
...
+		{
+			UpdateRecordedTips();
+			var found = false;
+			foreach (var tip in ProtipSaveStates)
+			{
+				if (tip.ID != ID) continue;
+				tip.Remembered = dontShowAgain;
+				found = true;
+				break;
+			}
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Learning/ProtipManager.cs]: Why not serialize as a dictionary?

File: UnityProject/Assets/Scripts/Learning/ProtipManager.cs [hunk 2]
```diff
@@ -39,13 +45,53 @@ public override void Awake()
...
+
+			SaveProtipSaveStates();
+		}
+
+		private void UpdateRecordedTips()
+		{
+			if (File.Exists(jsonPath) == false || File.ReadAllText(jsonPath).Length <= JSON_EMPTY_LIST) return;
+			var newList = JsonConvert.DeserializeObject<List<ProTipSaveData>>(File.ReadAllText(jsonPath));
+			ProtipSaveStates = newList;
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Learning/ProtipManager.cs]: As this comes from IO we can't guarantee the string is always deserializable.  `DeserializeObject()` might return null or throw (check documentation).

At the least, null check it:
```suggestion
			var newList = JsonConvert.DeserializeObject<List<ProTipSaveData>>(File.ReadAllText(jsonPath));
			ProtipSaveStates = newList ?? new List<ProTipSaveData>();
```

File: UnityProject/Assets/Scripts/Learning/UI/ProtipListUI.cs [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
-using UnityEngine.UI;
 
 namespace Learning
 {
 	public class ProtipListUI : MonoBehaviour
 	{
 		[SerializeField] private GameObject entryToSpawn;
 		[SerializeField] private Transform entryList;
+		[SerializeField] private Transform mainPage;
+		[SerializeField] private Transform settingsPage;
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Learning/UI/ProtipListUI.cs:11]: I don't like this - settings consumers shouldn't communicate directly with the settings UI - leave that up to the settings scripts.

File: UnityProject/Assets/Scripts/Learning/ProtipManager.cs (hunk 2 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Learning/ProtipManager.cs]: this causes an NRE when Rider suggested it

Comment (corp-0) [reply UnityProject/Assets/Scripts/Learning/ProtipManager.cs]: handle it wherever you use ``ProtipSaveStates``.

File: UnityProject/Assets/Scripts/Learning/UI/ProtipListUI.cs (hunk 3 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Learning/UI/ProtipListUI.cs:11]: what do you mean

Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Learning/UI/ProtipListUI.cs:11]: Apologies, I thought this was linking to the escape-menu settings page 🙂

**Review by NoooneyDude — APPROVED** [review]

**Pull request merged**